### PR TITLE
ENH add moment weight sums to output

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds the overall moment weight function normalization to fix a bug in mdet where this is ignored when combining shapes.

To do:

 - [x] tests

### CODE REVIEW

File: ngmix/gmix/gmix.py [hunk 1]
```diff
@@ -1239,15 +1239,13 @@ def get_weighted_moments_stats(ares):
...
-            res[n] = ares[n].copy()
-        elif n == "sums_cov":
+        if n in ["sums", "sums_cov"]:
             res[n] = ares[n].copy()
         else:
             res[n] = ares[n]
 
     res.update(
-        moments.make_mom_result(res["sums"], res["sums_cov"])
+        moments.make_mom_result(res["sums"], res["sums_cov"], res["wsum"])
```
Comment (beckermr) [inline ngmix/gmix/gmix.py:1248]:
```suggestion
        moments.make_mom_result(res["sums"], res["sums_cov"], mom_norm=res["wsum"])
```

File: ngmix/prepsfmom.py [hunk 2]
```diff
@@ -160,11 +160,11 @@ def _meas(self, obs, psf_obs, return_kernels):
         tot_var = np.sum(1.0 / obs.weight[msk])
 
         # run the actual measurements and return
-        mom, mom_cov = _measure_moments_fft(
+        mom, mom_cov, mom_norm = _measure_moments_fft(
             kim, kpsf_im, tot_var, eff_pad_factor, kernels,
             im_row - psf_im_row, im_col - psf_im_col,
         )
-        res = make_mom_result(mom, mom_cov)
+        res = make_mom_result(mom, mom_cov, mom_norm)
```
Comment (beckermr) [inline ngmix/prepsfmom.py:167]:
```suggestion
        res = make_mom_result(mom, mom_cov, mom_norm=mom_norm)
```

File: ngmix/tests/test_prepsfmom.py [hunk 3]
```diff
@@ -782,7 +782,7 @@ def test_moments_make_mom_result_flags():
     # neg flux
     _mom = mom.copy()
     _mom[5] = -1
-    res = make_mom_result(_mom, mom_cov)
+    res = make_mom_result(_mom, mom_cov, 1)
```
Comment (beckermr) [inline ngmix/tests/test_prepsfmom.py:785]:
```suggestion
    res = make_mom_result(_mom, mom_cov, mom_norm=1)
```

File: ngmix/tests/test_prepsfmom.py [hunk 4]
```diff
@@ -806,7 +806,7 @@ def test_moments_make_mom_result_flags():
         _mom_cov = mom_cov.copy()
         _mom_cov[4, i] = np.nan
         _mom_cov[i, 4] = np.nan
-        res = make_mom_result(mom, _mom_cov)
+        res = make_mom_result(mom, _mom_cov, 1)
```
Comment (beckermr) [inline ngmix/tests/test_prepsfmom.py:809]:
```suggestion
        res = make_mom_result(mom, _mom_cov, mom_norm=1)
```

File: ngmix/moments.py [hunk 5]
```diff
@@ -349,7 +349,7 @@ def g2mom(g1, g2, T):
     return Irr, Irc, Icc
 
 
-def make_mom_result(mom, mom_cov):
+def make_mom_result(mom, mom_cov, mom_norm=None):
```
Comment (esheldon) [inline ngmix/moments.py:352]: Is it true that for all the cases we use this function in `ngmix`, `mom` is actually `sums`?  In that case I think we should rename to `sums` for clarity

Comment (esheldon) [reply ngmix/moments.py:352]: Speaking for myself, this has caused me considerable confusion

Comment (esheldon) [reply ngmix/moments.py:352]: In that case, I think `mom_norm -> norm` or something

**Review by esheldon — CHANGES_REQUESTED** [review]

File: ngmix/moments.py (hunk 5 above)
Comment (beckermr) [reply ngmix/moments.py:352]: I can rename to sums. Codes are gonna break for a while due to that. We should use `sums_norm` I think.

File: ngmix/moments.py [hunk 6]
```diff
@@ -389,6 +392,7 @@ def make_mom_result(mom, mom_cov):
     res["flux"] = mom[mf_ind]
     res["mom"] = mom
     res["mom_cov"] = mom_cov
+    res["mom_norm"] = mom_norm if mom_norm is not None else np.nan
```
[the remark this answers is not in the document]
Comment (beckermr) [reply ngmix/moments.py:395]: Let's keep this as is and I'll fix this in the mdet repo.

[the remark this answers is not in the document]
Comment (beckermr) [reply ngmix/moments.py:395]: fixed in mdet!

**Comment by beckermr** [comment]
OK @esheldon This one is ready for another look. I renamed everything. Note that metadetect won't pass until we push a change to that repo. I'll do that once we release ngmix 2.1 which should be sometime today hopefully.

**Comment by esheldon** [comment]
Does this need the other PR to pass tests?

**Comment by esheldon** [comment]
sorry, getting confused on PRs

**Comment by esheldon** [comment]
OK, the mdet regression won't pass until both PRs are merged

**Review by esheldon — APPROVED** [review]

**Comment by beckermr** [comment]
Mdet won't pass until it is updated to use sums instead of mom. 

So ignore that test for now.

File: CHANGES.md [hunk 7]
```diff
@@ -1,10 +1,15 @@
 ## v2.1.0
 
+### bug fixes
+
+    - Fixed 1/area normalization for all moments fields in `GaussMom` results.
+
```
Comment (beckermr) [inline CHANGES.md:6]:
```suggestion
### changes

    - Changed "mom*" fields in the moments outputs to "sums*". 
```

**Pull request merged**